examples/step-22/step-22.cc

```diff
@@ -996,21 +996,24 @@ namespace Step22
   // rectangle four times in the first coordinate direction. To limit the
   // scope of the variables involved in the creation of the mesh to the range
   // where we actually need them, we put the entire block between a pair of
-  // braces:
+  // braces. In order to allow both 2d and 3d coordinates, we make use of the
+  // combination of std::make_pair and std::get functions:
   template <int dim>
   void StokesProblem<dim>::run()
   {
     {
       std::vector<unsigned int> subdivisions(dim, 1);
       subdivisions[0] = 4;
 
-      const Point<dim> bottom_left = (dim == 2 ?                //
-                                        Point<dim>(-2, -1) :    // 2d case
-                                        Point<dim>(-2, 0, -1)); // 3d case
+      const auto bottom_left_2d_or_3d =
+        std::make_pair(Point<2>(-2, -1),     // 2d case
+                       Point<3>(-2, 0, -1)); // 3d case
+      const Point<dim> bottom_left = std::get<dim - 2>(bottom_left_2d_or_3d);
```

Comment:
Since I changed the `Assert` statements in `Point<dim>` constructors into `static_assert`, the original statement
```c++
Point<dim> bottom_left = (dim == 2 ? Point<dim>(-2, -1) :  Point<dim>(-2, 0, -1));
```
fails to compile for both `dim == 2` and `dim == 3`. The neatest possible way I found around is using `std::make_pair` or `std::make_tuple`. Do you think this is an acceptable and readable modification?

Reply:
Oh, I see. No, this isn't acceptable -- we use this idiom all over the place, not just in the library or the tutorial, but many user programs as well. All of these would now stop to compile. So that can't work.

You'll have to find a different way to dealing with this than the `static_assert` in the constructor. Sorry :-(